Library/Homebrew/language/go.rb

```diff
@@ -12,7 +12,9 @@ module Go
     # building Go software.
     # The resource names should be the import name of the package,
     # e.g. `resource "github.com/foo/bar"`.
+    sig { params(resources: T::Array[Resource], target: T.any(String, Pathname)).void }
     def self.stage_deps(resources, target)
+      # odeprecated "Language::Go::stage_deps", "or request upstream to migrate to Go modules"
```

Comment:
Is it worth saying this is a future intended deprecation (in a comment above) otherwise it might be confusing for people combing through deprecations.

Reply:
@issyl0 This is how we normally handle "deprecate this next major/minor release" deprecations.

Reply:
Ah, thanks for the clarification!

Reply:
Maybe we can reword this somehow? Otherwise we end up with

```
use or request upstream to migrate to Go modules instead.
```

Reply:
I was originally either going to leave it empty or thinking of just `Go modules` (i.e. `use Go modules instead`)

I partly mixed in another related message from: 
https://github.com/Homebrew/brew/blob/9ae51618b96e6544f08e9ef9c0a34cd6576eb83d/Library/Homebrew/rubocops/text.rb#L117

Reply:
I can do any extra updates to deprecation message in a separate PR and also add comment for `go_resource`s. May just go with something like:
```suggestion
      # odeprecated "Language::Go::stage_deps", "Go modules"
```

As I understand, vendoring would also refer to older solution with vendor directory and GOPATH, which we are also trying to get rid of.

Reply:
> I can do any extra updates to deprecation message in a separate PR and also add comment for `go_resource`s. May just go with something like:
> 
> In case of existing message, as I understand, vendoring would also refer to older solution with vendor directory and GOPATH, which we are also trying to get rid of.

---

I guess another option is to recommend: `Language::Go::stage_deps --> resources.each { |r| (target/r.name).install r}` and `go_resource --> resource`. Though, this means the formula is still using an old approach.

I have used these to get rid of all usage of `go_resources` in Homebrew/core.